### ISSUE TITLE
RT crashes when select the folder destination from queue screen

### ISSUE THREAD

**Issue by jazziman** [Beep6581/RawTherapee#3693]
RT crashes when, from the development queue screen, I try to select the folder destination of the developed files.

[crash-report.txt](https://github.com/Beep6581/RawTherapee/files/775844/crash-report.txt)

**Comment by TooWaBoo** [comment]
This is known bug on Mac. It happens when the folder is set to "None".
see: https://github.com/Beep6581/RawTherapee/issues/3672

**Comment by jazziman** [comment]
Thanks TooWaBoo, but I do not understand how and where I can correct the error. What is the file to be corrected?

Thanks

**Comment by TooWaBoo** [comment]
You have to wait for a new version. I don't know anything about RT on the Mac.
Perhaps @Partha1b can help you.

**Comment by heckflosse** [comment]
@jazziman You can try to enter the path in options file. Search for `PathFolder` in options file. Once the path is filled correctly it could be possible to change it in rt as well.
Close rt before editing the options file!

**Comment by jazziman** [comment]
I see the others strings of output of file "options" situated in User/Library/Application Support/Rawtherapee.
In [External Editor] “PhotoshopDir=“ and in [Output] “PathFolder=” I added a simple: /Users
It’s all ok, in the queue tab and in preference, it’s possible to select the destination without crash.

————————

[External Editor]
EditorKind=1
GimpDir=
PhotoshopDir=/Users
CustomEditor=

———————-

[Output]
Format=jpg
JpegQuality=92
JpegSubSamp=2
PngCompression=6
PngBps=8
TiffBps=16
TiffUncompressed=true
SaveProcParams=true
FormatBatch=jpg
JpegQualityBatch=92
JpegSubSampBatch=2
PngCompressionBatch=6
PngBpsBatch=8
TiffBpsBatch=16
TiffUncompressedBatch=true
SaveProcParamsBatch=true
PathTemplate=%p1/converted/%f
PathFolder=/Users
AutoSuffix=true
ForceFormatOpts=true
SaveMethodNum=0
UsePathTemplate=false
LastSaveAsPath=
OverwriteOutputFile=false
TunnelMetaData=true

**Comment by jazziman** [comment]
...certainly for a code programmer, it will not be the correct way, but... It works! :-)

**Comment by heckflosse** [comment]
@jazziman I would prefer to solve it in code but I don't have access to a Mac

**Comment by jazziman** [comment]
@heckflosse Can I put my time to experiment and communicate to you the results, but I do not understand anything of programming code. You should try to give me some strings and I can do the tests.

**Comment by heckflosse** [comment]
@jazziman Ok, great, let's start with `folder destination from queue screen`. The code is in file
`rtgui/batchqueuepanel.cc` line 116 to 118

```
    if (Glib::file_test (options.savePathFolder, Glib::FILE_TEST_IS_DIR)) {
        outdirFolder->set_current_folder (options.savePathFolder);
    }
```
That means, if the path in options file is empty or a non existing path, the path won't be set, which leads to the crash when trying to select a path. On Windows this works fine:

```
    if (Glib::file_test (options.savePathFolder, Glib::FILE_TEST_IS_DIR)) {
        outdirFolder->set_current_folder (options.savePathFolder);
    } else {
        outdirFolder->set_current_folder (Glib::ustring());
    }
```
because it magically sets the folder rt was started from.
Unfortunately that does not seem to work on mac.
So we have to set a path there which exists. 
I don't know which environment variables exist on OSX but if `HOME` exists then this:
```
    if (Glib::file_test (options.savePathFolder, Glib::FILE_TEST_IS_DIR)) {
        outdirFolder->set_current_folder (options.savePathFolder);
    } else {
        char *homeDir = getenv("HOME");
        if(homeDir) {
            outdirFolder->set_current_folder (homeDir);
        }
    }
```
should work.

**Comment by heckflosse** [comment]
@Partha1b could you also try the solution from above post?

**Comment by Floessie** [comment]
@heckflosse There's a [Glib::get_home_dir()](https://developer.gnome.org/glibmm/stable/group__MiscUtils.html#ga9412ea70c7fea058c03211dac318f8e6).

**Comment by heckflosse** [comment]
@Floessie great, even better :+1:

**Comment by Partha1b** [comment]
@heckflosse and @Floessie Yes, the above works!  😄 

I was approaching it differently and not using another else block which did not work.

I think that instead of using a similar fix for PS folder, we should default to the standard installation folder (just like we do for "official GIMP"). Anyone who installs it in a non-standard folder can use the command line option. What do you think?

Thanks,
Partha

**Comment by jazziman** [comment]
I'm sorry but just as I had anticipated to you, because I not am aware of the bases of programming I could not do anything. By the time I solved the problem by simply placing a "/" in the empty fields of the "options" file that I've described in previous posts. The file "rtgui/batchqueuepanel.cc" is not resident on Mac. I remain available for other tests on my Mac.

**Comment by Partha1b** [comment]
@jazziman This file is part of the RT source code.

**Comment by jazziman** [comment]
@Partha1b  ...source code from this: http://rawtherapee.com/shared/source/

**Comment by heckflosse** [comment]
Should be fixed with 6a3baa3. Closing.
